SonarQube.TeamBuild.PreProcessor/SonarWebService.cs

```diff
@@ -233,6 +236,106 @@ public bool TryDownloadEmbeddedFile(string pluginKey, string embeddedFileName, s
+            }
+            else
+            {
+                serverVersion = new Version(contents);
+            }
+        }
+
+        private IDictionary<string, string> GetPropertiesOld(string projectId)
+        {
+            string ws = GetUrl("/api/properties?resource={0}", projectId);
+            this.logger.LogDebug(Resources.MSG_FetchingProjectProperties, projectId, ws);
+            var contents = this.downloader.Download(ws);
+
+            var properties = JArray.Parse(contents);
+            var result = properties.ToDictionary(p => p["key"].ToString(), p => p["value"].ToString());
+
+            // http://jira.sonarsource.com/browse/SONAR-5891 or when C# plugin is not installed
+            if (!result.ContainsKey("sonar.cs.msbuild.testProjectPattern"))
+            {
+                result["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
+            }
+
+            return result;
+        }
+
+        private IDictionary<string, string> GetProperties63(string projectId)
+        {
+            string ws = GetUrl("/api/settings/values?component={0}", projectId);
+            this.logger.LogDebug(Resources.MSG_FetchingProjectProperties, projectId, ws);
+            var contents = this.downloader.Download(ws);
+
+            Dictionary<string, string> settings = new Dictionary<string, string>();
+            var settingsArray = JObject.Parse(contents).Value<JArray>("settings");
+            foreach (var t in settingsArray)
+            {
+                getPropertyValue(settings, t);
+            }
+
+            // http://jira.sonarsource.com/browse/SONAR-5891 or when C# plugin is not installed
+            if (!settings.ContainsKey("sonar.cs.msbuild.testProjectPattern"))
+            {
+                settings["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
+            }
+
+            return settings;
+        }
+
+        private void getPropertyValue(Dictionary<string, string> settings, JToken p)
+        {
+            if (p.Value<Boolean>("inherited"))
+            {
+                return;
+            }
+
+            string key = p["key"].ToString();
+            if (p["value"] != null)
+            {
+                string value = p["value"].ToString();
+                settings.Add(key, value);
+            }
+            else if (p["fieldValues"] != null)
+            {
+                multivalueToProps(settings, key, (JArray)p["fieldValues"]);
+            }
+            else if (p["values"] != null)
+            {
+                JArray array = (JArray)p["values"];
+                string value = string.Join(",", array.Values<string>());
+                settings.Add(key, value);
+            }
+            else
+            {
+                throw new ArgumentException("Invalid property");
+            }
+        }
+
+        private void multivalueToProps(Dictionary<string, string> props, string settingKey, JArray array)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename method 'multivalueToProps' to match camel case naming rules, consider using 'MultivalueToProps'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS100)